packages/core/src/application_init.ts

```diff
@@ -68,6 +68,9 @@ export class ApplicationInitStatus {
         const initResult = this.appInits[i]();
         if (isPromise(initResult)) {
           asyncInitPromises.push(initResult);
+        } else if (isObservable(initResult)) {
+          asyncInitPromises.push(new Promise<void>(
+              (resolve, reject) => initResult.subscribe({complete: resolve, error: reject})));
```

Comment:
Minor suggestion (for better readability):
```suggestion
          const observableAsPromise = new Promise<void>((resolve, reject) => {
            initResult.subscribe({complete: resolve, error: reject});
          });
          asyncInitPromises.push(observableAsPromise);
```